### DISCUSSION TITLE
json contains method -- prevent throwing error exceptions

### DISCUSSION THREAD

**Discussion by mohammedalasmar** [nlohmann/json#3666] [Q&A] [answered]
I have a json payload represented as std::string

```JS
std::string payload = 
[
  {
    "type": 22,
    "msg": {
      "value": 0
    }
  }
]
 ```

I use the parse method to create a JSON object, as follows:
```JS
 nlohmann::json rcvdJSON =  nlohmann::json::parse(payload);
 ```

Now, I strat reading the values from this object:

Example 1:
```JS
 int msgType = rcvdJSON[0]["type"];
 ```
   
Example 2:
```JS
int val = rcvdJSON[0]["msg"]["value"];
 ```

My question is that if the received JSON doesn't have the expected structure then the lib will through an error. I want to prevent this from happening by checking if that object contains what I'm trying to read. I'm trying to use this method: ".contains" but I'm not sure how to use it in the above two examples.

**Comment by nlohmann** [comment, marked as answer]
Did you check the [documentation](https://json.nlohmann.me/api/basic_json/contains/)?

**Reply by mohammedalasmar** [reply]
yes, I saw example 3 but I couldn't apply it to my examples

**Reply by nlohmann** [reply]
If you want to make sure that `msg` exists in the first element of the array, do this:

```cpp
if (rcvdJSON[0].contains("msg") { ... }
```

If you also want to make sure there is a "value" key:

```cpp
if (rcvdJSON[0].contains("msg") && rcvdJSON[0].contains["msg"]("value") { ... }
```

**Reply by mohammedalasmar** [reply]
Many thanks! 


I just tried: 


```JS
rcvdJSON[0].contains["msg"]("value") 
 ```
and the syntax seems to be wrong?

**Reply by nlohmann** [reply]
Yes, contains needs normal parentheses.

**Reply by falbrechtskirchinger** [reply]
```cpp
if (rcvdJSON[0].contains("msg") && rcvdJSON[0]["msg"].contains("value") {
    int value = rcvdJSON[0]["msg"]["value"];
}
```

**Reply by nlohmann** [reply]
Sorry, my bad.
